include/mxnet/c_api_common.h

```diff
@@ -22,8 +22,8 @@
  * \file c_api_error.h
  * \brief Error handling for C API.
  */
-#ifndef MXNET_C_API_C_API_COMMON_H_
-#define MXNET_C_API_C_API_COMMON_H_
+#ifndef MXNET_C_API_COMMON_H_
+#define MXNET_C_API_COMMON_H_
 
 #include <dmlc/base.h>
 #include <dmlc/logging.h>
```

Comment:
Do you need all these functions in the public header files, including `MXAPIThreadLocalEntry`, `CopyAttr`, etc?

Reply:
Do you mind pointing out how horovod will use these APIs?

Reply:
@eric-haibin-lin I agree `MXAPIThreadLocalEntry`, `CopyAtt` are not need in error handling. I don't know the historical reason of adding them to this header file in the first place. My change was following the rationale of making least change to existing structure.

I could extract the error handling functions from `c_api_common.h` into `c_api_error.h` and only export `c_api_error.h` to include. Does that sound like a better approach to you?